pkg/plugins/resources/cargopackage/spec.go

```diff
@@ -1,18 +1,15 @@
 package cargopackage
 
 import (
+	"github.com/updatecli/updatecli/pkg/plugins/utils/cargo"
 	"github.com/updatecli/updatecli/pkg/plugins/utils/version"
 )
 
 // Spec defines a specification for a "dockerimage" resource
 // parsed from an updatecli manifest file
 type Spec struct {
-	// [S][C] IndexUrl specifies the url of the index to use to check version
-	// If set, package version will be checked against the API
-	IndexUrl string `yaml:",omitempty"`
```

Comment:
This is a breaking change that could easily be avoided with a deprecation approach
We try to carefully introduce new "user" facing parameters as we want to avoid as much as possible breaking changes.

Something like 

```
`IndexUrl string `yaml:",omitempty" jsonschema:"-"``
```
Means that the filed won't appear in the generated jsonschema and so will return an error if the indexurl parameter is used.

Then in the new object you could have something like 

```
if spec.IndexURL != "" {
  switch spec.Registry.URL != {
    case true:
      logrus.Warningf("Registry.URL and IndexURL are mutually exclusive, unset indexurl")
      spec.IndexURL = ""
    case false:
      logrus.Warningf("indexurl is deprecated in favor of registry.url")
      spec.IndexURL = ""
  }
}

By emptying the deprecated field in favor of the new field, the following command will automatically update the old manifest

`updatecli manifest upgrade`